u_parrots.go

```diff
@@ -527,6 +527,442 @@ func utlsIdToSpec(id ClientHelloID) (ClientHelloSpec, error) {
 				}},
 			},
 		}, nil
+	case HelloEdge_85:
+		return ClientHelloSpec{
+			CipherSuites: []uint16{
+				GREASE_PLACEHOLDER,
+				TLS_AES_128_GCM_SHA256,
+				TLS_AES_256_GCM_SHA384,
+				TLS_CHACHA20_POLY1305_SHA256,
+				TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+				TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+				TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+				TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+				TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,
+				TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
+				TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+				TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+				TLS_RSA_WITH_AES_128_GCM_SHA256,
+				TLS_RSA_WITH_AES_256_GCM_SHA384,
+				TLS_RSA_WITH_AES_128_CBC_SHA,
+				TLS_RSA_WITH_AES_256_CBC_SHA,
+			},
+			CompressionMethods: []uint8{
+				0x0, // no compression
+			},
+			Extensions: []TLSExtension{
+				&UtlsGREASEExtension{},
+				&SNIExtension{},
+				&UtlsExtendedMasterSecretExtension{},
+				&RenegotiationInfoExtension{
+					Renegotiation: RenegotiateOnceAsClient,
+				},
+				&SupportedCurvesExtension{
+					Curves: []CurveID{
+						GREASE_PLACEHOLDER,
+						X25519,
+						CurveP256,
+						CurveP384,
+					},
+				},
+				&SupportedPointsExtension{
+					SupportedPoints: []uint8{
+						0x0, // pointFormatUncompressed
+					},
+				},
+				&SessionTicketExtension{},
+				&ALPNExtension{
+					AlpnProtocols: []string{
+						"h2",
+						"http/1.1",
+					},
+				},
+				&StatusRequestExtension{},
+				&SignatureAlgorithmsExtension{
+					SupportedSignatureAlgorithms: []SignatureScheme{
+						ECDSAWithP256AndSHA256,
+						PSSWithSHA256,
+						PKCS1WithSHA256,
+						ECDSAWithP384AndSHA384,
+						PSSWithSHA384,
+						PKCS1WithSHA384,
+						PSSWithSHA512,
+						PKCS1WithSHA512,
+					},
+				},
+				&SCTExtension{},
+				&KeyShareExtension{
+					KeyShares: []KeyShare{
+						{
+							Group: GREASE_PLACEHOLDER,
+							Data: []byte{
+								0,
+							},
+						},
+						{
+							Group: X25519,
+						},
+					},
+				},
+				&PSKKeyExchangeModesExtension{
+					Modes: []uint8{
+						PskModeDHE,
+					},
+				},
+				&SupportedVersionsExtension{
+					Versions: []uint16{
+						GREASE_PLACEHOLDER,
+						VersionTLS13,
+						VersionTLS12,
+						VersionTLS11,
+						VersionTLS10,
+					},
+				},
+				&GenericExtension{
+					// Certificate Compression:
+					// https://tools.ietf.org/html/draft-ietf-tls-certificate-compression-10
+					Id: 27,
+					Data: []byte{
+						2, 0, 2,
+					},
+				},
```

Comment:
Can you implement this with https://github.com/getlantern/utls/blob/master/u_tls_extensions.go#L699 ?

Reply:
The reason these are generic extensions is because I used `FingerprintClientHello` to generate these specs and `FingerprintClientHello` instantiates these extensions using `GenericExtension`.  I will be adding the tool I used to generate these to flashlight (or maybe here, I'm not sure yet where it fits best, but that's another discussion).  In the interest of keeping that tool useful (and just improving `FingerprintClientHello`), I think the best route would be to have `FingerprintClientHello` return the proper extension types rather than `GenericExtension`.

Reply:
What do you think?

Reply:
Ah yeah so my thought is that if this is roughly a one-off process (or something we have to do manually once in a while), having to do a little struct munging to replace `GenericExtension`s with specific ones might not be too bad. That being said, I don't think there's really any reason not to implement them in `FingerprintClientHello`.

Mostly the reason I mention it here is that having had some back and forths with Sergey, my suspicion is that he'll want these to be specific

Reply:
> having to do a little struct munging to replace GenericExtensions with specific ones might not be too bad

I think I'd just rather have the process specified as code than instructions in a README somewhere.  `FingerprintClientHello` seems like a logical place to encode the transformation.  I can look into making that happen.

Reply:
https://github.com/getlantern/utls/pull/12